### ISSUE TITLE
Integrate export function

### ISSUE THREAD

**Issue by ChristineDomgoergen** [wmde/mitmachen#62]
Integrate an export function in the tool that makes the export of both data sets, page views and search terms possible in csv format

**Comment by darionewmonday** [comment]
2 different CVS files. 
1 static title per file. We will discuss a different alternative. @dev-ckln

**Comment by dev-ckln** [comment]
Its done, please check by clicking on the csv field.
You can even download a subset of data by first searching for something and then clicking on the CSV button which will download the filter data only.

**Comment by darionewmonday** [comment]
We said 2 CSVs, now it´s only one. 
We need 2 export widgets. @dev-ckln

**Comment by dev-ckln** [comment]
Single csv button can export any group of data. If you filter data via search and then click on CSV button, you only get the filtered data.

**Comment by darionewmonday** [comment]
I understand that. What I mean is, we would like to have 2 export options in order to download either the search-terms data OR the page stats. 
Would it be possible?

**Comment by dev-ckln** [comment]
Yes that's what I meant. You can filter data via search and then click on CSV button to download the filtered data.

**Comment by darionewmonday** [comment]
The CSV file doesn´t include the data of the page views. This is what I meant, we need BOTH data and both exports. @dev-ckln

**Comment by dev-ckln** [comment]
Solution : 
The current CSV button will be used to download the complete page stats.
And for the search terms data, when user will enter any keyword in the search input box then filtration will get performed and a new button will appear beside the search input box to download the search terms data.

Let me know if it is fine.

**Comment by darionewmonday** [comment]
@ChristineDomgoergen what do you think?

**Comment by ChristineDomgoergen** [comment]
@darionewmonday 
We really need all the search terms data per csv to do a proper analysis. Can't we include two csv export buttons?

**Comment by dev-ckln** [comment]
<img width="1403" alt="Screen Shot 2020-03-12 at 2 14 11 PM" src="https://user-images.githubusercontent.com/48859836/76503381-cfdf0800-646b-11ea-8163-571911fe39f3.png">

**Comment by dev-ckln** [comment]
Currently given downloads buttons are working as decsribe below:

PAGE VIEW CSV : Button will download data where type is subcategoryarticle.
SEARCH TERM DATA CSV: Button will download data where type is Search.

Both of the button has no linking with the graph provided below.

Now you are requested to explain your concern in detail, so that without any further delay we can provide the same.

**Comment by darionewmonday** [comment]
Can you please simply move the button to download the page views CSV down to the column graph pls? https://prnt.sc/rgup6b

@dev-ckln

**Comment by dev-ckln** [comment]
Moving of button is not possible reason being, Those buttons are not explicitly made by me in HTML, they are a part of DataTable, and automatically made as part of table shown so moving the button far below the table is not possible.

**Comment by darionewmonday** [comment]
Allright, so maybe we can solve like this:
1) In the table above, leave only the button "search term data blabla..."
2) in the widget below, ADD "Download: "Page view ..."

ok like this? @dev-ckln

**Comment by dev-ckln** [comment]
I have gone through with it and the conclusion is, 

Both the buttons are a single unit and dynamically created with the table. So can't move either one, they can move together only.

**Comment by darionewmonday** [comment]
ok then let´s leave it as it is now.

**Comment by darionewmonday** [comment]
It doesn´t work. Both CSV show the same data set, pls check and fix it. @dev-ckln

**Comment by dev-ckln** [comment]
well, i have tried again and both button exporting different data as per below detail:

seitenaufrufe: Downloading data with type = subcategoryarticle

suchbegriffe: Downloadig data with type = Search

Please eplain more about your concern , and let me know if you need detailig with video.

**Comment by darionewmonday** [comment]
In the exported data, some letters are not encoded:
e.g.
SportausrÃ¼stung 
instead of 
Sportausrüstung

@dev-ckln

**Comment by darionewmonday** [comment]
The exported data don´t match the filtered data
Check the screenshot against this exel https://docs.google.com/spreadsheets/u/1/d/1ZJxiAeY1NqtcAM2sujgVBxEQRE1Ay9AjYUQ0_7pVFag/edit?urp=gmail_link#gid=1999516879


![Screenshot_2020-04-15 Mitmachen in Articles(1)](https://user-images.githubusercontent.com/53903644/79423314-afe8ba00-7fbe-11ea-8fb1-3dab2e3b53c9.png)

@dev-ckln

**Comment by dev-ckln** [comment]
Data filteration is working, 

SEITENAUFRUFE : Download the data where Type = subcategoryarticle
SUCHBEGRIFFE   : Download the data where Type = search

For Date range 11.03.20 to 15.04.20 filtered data and downloaded data is accurate, please match the data accordingly.

**Comment by dev-ckln** [comment]
Please review the list of FINAL changes.

There are two buttons : Page View , Search Term Data

- On clicking the PAGE VIEW button. The downloaded CSV will contain the following columns only.
**Date | Count**
10 APR | 4
11 APR | 5
Total View : 9
Note : The above data will be the same which displays in the graph view.

--------------------------

- On clicking on SEARCH DATA, the CSV will contain following fields.
**ID | TYPE | TITLE | DATE | COUNT**

Note : It will include the both TYPES of data. (Search and subcategoryarticle)
--------------------

To make this work, we need to remove the data filter from the top of the graph. There should be a single date filter on the top which will filter the content for both table and graph.
Let me know if you have any concern with it.

-------------------
Also the CSV names would be Pageview.csv and Searchdate.csv. We are going to update this too.
-------------------

At present there is some issue with chrome while translating the content, Some of the numbers are changing with letters. We are checking the same as well.

-------------------

We need your confirmation on the above points before doing them, if you have any issue let us know now otherwise we won't be able to rework on the same.

**Comment by darionewmonday** [comment]
@ChristineDomgoergen can you please check this comment and get back to me per email or telephone is anything is not clear?

My comment:
Why is "Subcategory" going to be changed with "Page views" ? @ChristineDomgoergen also here your input is helpful.

thanks to you both
